.github/workflows/create-xray-test-plan-and-test-execution.yml

```
@@ -0,0 +1,234 @@
+    runs-on: ubuntu-22.04
+    outputs:
+      test_plan_key: ${{ steps.get_test_plan_key.outputs.test_plan_key }}
+      test_plan_id: ${{ steps.get_test_plan_id.outputs.test_plan_id }}
+      test_execution_key: ${{ steps.get_test_execution_key.outputs.test_execution_key }}
+      test_execution_id: ${{ steps.get_test_execution_id.outputs.test_execution_id }}
+
+    steps:
+      - name: Generate Xray Token
+        id: generate_xray_token
+        run: |
+          token_response=$(curl -H "Content-Type: application/json" -X POST --data "{\"client_id\": \"${{ secrets.xray_client_id }}\", \"client_secret\": \"${{ secrets.xray_client_secret }}\"}" "https://xray.cloud.getxray.app/api/v1/authenticate")
+          xray_token=$(echo "$token_response" | sed -n 's/.*"\(.*\)".*/\1/p')
+          echo "xray_token=$xray_token" >> $GITHUB_OUTPUT
+        shell: bash
+
+      - name: Create or Get the TestPlanKey
+        id: get_test_plan_key
+        run: |
+          # Use a GraphQL query to get all test plans matching the JQL
+          graphql_query='{
+          "query":"query GetTestPlans($jql: String, $limit: Int!) { getTestPlans(jql: $jql, limit: $limit) { total results { issueId jira(fields: [\"summary\", \"key\"]) } } }",
+          "variables":{"jql": "project = MON","limit": 100}}'
+
+          # Execute the GraphQL query and parse the results
+          response=$(curl -H "Content-Type: application/json" -X POST -H "Authorization: Bearer ${{ steps.generate_xray_token.outputs.xray_token }}" --data "$graphql_query" "https://xray.cloud.getxray.app/api/v2/graphql")
+
+          test_plan_key=''
+          test_plans=$(echo "$response" | jq -c '.data.getTestPlans.results[].jira')
+
+          echo "Those are the existent TPs: $test_plans"
+
+          # Determine the summary's prefix
+          summary_prefix=''
+          if [[ ${{ github.event_name }} == 'schedule' ]]; then
+            summary_prefix="NIGHTLY"
+          else
+            echo "The github_ref_name is: $GITHUB_REF_NAME"
+            case "$GITHUB_REF_NAME" in
+              hotfix*)
+                summary_prefix="HOTFIX"
+                ;;
+              release*)
+                summary_prefix="RLZ"
+                ;;
+            esac
+          fi
+
+          input_summary="$summary_prefix OSS ${{ inputs.major_version }}"
+          echo "The summary to search is: $input_summary"
+
+          # Extract the key of the existent test plan
+          while read row; do
+          summary=$(echo "$row" | jq -r '.summary')
+          if [[ "$summary" == "$input_summary" ]]; then
+            test_plan_key=$(echo "$row" | jq -r '.key')
+            echo "The test_plan_key is $test_plan_key and the summary is $summary"
+            break
+          fi
+          done <<< "$test_plans"
+
+          echo "The test plan key for now is: $test_plan_key"
+
+          # If no matching test plan was found, create one
+          if [ -z "$test_plan_key" ]; then
```

```suggestion
          if [[ -z "$test_plan_key" ]]; then
```
consistency with other line like +145